spark/src/main/scala/io/delta/sql/DeltaSparkSessionExtension.scala

```diff
@@ -91,6 +91,23 @@ class DeltaSparkSessionExtension extends (SparkSessionExtensions => Unit) {
     extensions.injectResolutionRule { session =>
       PreprocessTimeTravel(session)
     }
+    // Inject MaybeFallbackV1Connector to fallback SparkTable to DeltaTableV2 for unsupported
+    // operations (e.g., streaming, write operations) in kernel-spark.
+    try {
+      // scalastyle:off classforname
+      val constructor = Class.forName("io.delta.kernel.spark.MaybeFallbackV1Connector")
```

Comment:
Shall we create a new `KernelSparkSessionExtension` which extends `DeltaSparkSessionExtension`?